docs/src/quickstart.md

```diff
@@ -160,6 +161,51 @@ plot_energy(idata);
 gcf()
 ```
 
+### Additional information in Turing.jl
+In the above `from_mcmcchains` call, we did not specify the `prior`, `posterior_predictive`, nor the `log_likelihoods`. This section describes how one would go about obtaining these from the `Model` and `Chains`, which we can then pass to `from_mcmcchains` to construct the [`InferenceData`](@ref).
+
+To sample from the prior, one simply calls `sample` but with the `Prior` sampler:
+```@example quickstart
+prior = sample(param_mod, Prior(), nsamples)
+```
+Obtaining the predictive posterior is a matter of first instantiating a "predictive model", i.e. a `Model` but with the observations set to `missing`, and then call `predict` on the predictive model and the inferred posterior:
+```@example quickstart
+# Instantiate the predictive model
+param_mod_predict = turing_model(
+    J, fill(missing, length(y)), σ
+)
+# and then sample!
+posterior_predictive = predict(param_mod_predict, turing_chns)
+```
+And to extract the elementwise log-likelihoods, which is useful if you want to compute metrics such as [`loo`](@ref),
+```@example quickstart
+loglikelihoods = elementwise_loglikelihoods(param_mod, turing_chns)
+```
+This can then be included in the `from_mcmcchains` call from above:
+```@example quickstart
+# Convert into format compatible with ArviZ.jl
+using LinearAlgebra
+loglikelihoods_arr = permutedims(cat(values(loglikelihoods)...; dims = 3), (2, 1, 3))
```

Comment:
Ah, yeah that's a good point! I didn't think about the ordering as I at the time I was mainly interested in `loo`. For the tutorial, maybe we should just match the keys of `loglikelihoods` with the predictive posterior `Chains`, e.g.
```julia
loglikelihoods_vals = getindex.(Ref(loglikelihoods), string.(keys(posterior_predictive)))
loglikelihoods_arr = permutedims(cat(loglikelihoods_vals...; dims = 3), (2, 1, 3))
```

> I'm curious, why does Turing return a Dict here instead of a Chains?

A couple of reasons. 
1. `elementwise_loglikelihood` is part of `DynamicPPL` which doesn't use MCMCChains.jl, hence we wanted to use a structure which was independent of MCMCChains.jl. But of course we could just wrap this in Turing.jl, which would then return a `Chains`, which brings me to the second point.
2. I'd argue that the elementwise loglikelihoods doesn't quite fit into a `Chains` structure, _as it is now_. We could of course just add fields `y[1]_loglikelihood` or something, but that will make things cluttered quickly + we'll have to implement something hacky so the user can access to the elementwise loglikelihoods directly from the chain (e.g. `elementwise_loglikelihoods(c::Chains)` which finds the corresponding fields using `Regex`). The _correct_ approach I think would be something similar to `InferenceData` where the loglikelihoods is a "special" field in the chain. But at the moment, this sort of goes against the approach taken in MCMCChains.jl where no fields are treated in a special manner; it's just a matter of whether or not it's a `parameter` or `internal`. And if we made this an `internal` field, we would still have to wrap it in some other container to map observation variable name to the loglikelihood.

Thinking about it again, maaaybe it's not such a bad idea to return a chain where the only fields present are the loglikelihoods (I was initially thinking of returning a chain from `predict` with both the sampled values and the loglikelihoods), e.g. `y[1]` now represents the loglikelihoods. I'm not a huge fan though as it could be confusing to look at a chain with the keys being the same as in the chain obtained from `sample` but now representing something different. So I think I'm leaning towards something along the lines as I mentioned in (2) above.

Reply:
Thanks! That helps a lot. For me, it would be handy of log-likelihoods were not returned in a generic data structure like a `Dict`, because then I can't infer anything about them from the type, hence the need below to manually reshape, which is annoying and could be error-prone for the user.

> Thinking about it again, maaaybe it's not such a bad idea to return a chain where the only fields present are the loglikelihoods (I was initially thinking of returning a chain from predict with both the sampled values and the loglikelihoods), e.g. y[1] now represents the loglikelihoods. I'm not a huge fan though as it could be confusing to look at a chain with the keys being the same as in the chain obtained from sample but now representing something different. So I think I'm leaning towards something along the lines as I mentioned in (2) above.

Here are my general thoughts. First, I don't think it needs to be confusing to have log-likelihoods in a data structure meant for random samples. After all, because the log-likelihoods are deterministically computed from random samples and constants, the resulting log-likelihoods are in fact random samples. The big downside I see is, yeah, I think it could be confusing for the log-likelihoods to have the same name as the variable, but since the user is explicitly calling an `elementwise_loglikelihoods` function, and the values aren't just being injected into another function call, that doesn't seem like a big deal if documented. It's just that the user really should never merge such a `Chains` object with a posterior or prior sample. I'd have to think more about the (2) approach.

Reply:
>  because then I can't infer anything about them from the type

So the return-value from `elementwise_loglikelihood` is always going to be the same type, i.e. `Dict{String, Matrix{Float64}}`, so the processing that we do in this readme could be automated. I get what you're saying though, and I agree.

> First, I don't think it needs to be confusing to have log-likelihoods in a data structure meant for random samples. 

Yeah I agree with this. My issue is that I don't think the _currrent_ `MCMCChains.Chains` is suitable for also having the `loglikelihoods` together with the samples :confused: 

>  It's just that the user really should never merge such a Chains object with a posterior or prior sample. I'd have to think more about the (2) approach.

The thing is I'm worried that this could happen at some point though, e.g. it seems natural to allow returning the loglikelihoods from `predict` in addition to the predicted "observations".